### ISSUE TITLE
[STORM-1870] Allow FluxShellBolt/Spout set custom "componentConfig" via yaml.

### ISSUE THREAD

**Issue by jira-importer** [apache/storm#5653]
<p>FluxShellBolt/Spout should have option to provide custom config option when importing topology from YAML file.</p>

<p>We use this to provide custom configuration for example to our python RabbitMQ bolts. (Passing strings and list of strings).</p>

<p>We are using this code:</p>
<div class="code panel" style="border-style: solid;border-width: 1px;"><div class="codeHeader panelHeader" style="border-bottom-width: 1px;border-bottom-style: solid;"><b>FluxShellBolt.java</b></div><div class="codeContent panelContent">
<pre class="code-java"><span class="code-keyword">public</span> <span class="code-keyword">class </span>FluxShellBolt <span class="code-keyword">extends</span> ShellBolt <span class="code-keyword">implements</span> IRichBolt {
    
    <span class="code-comment">//... (<span class="code-keyword">rest</span> of the class)
</span>
    <span class="code-keyword">public</span> void addComponentConfig(<span class="code-object">String</span> key, <span class="code-object">Object</span> value) {
<span class="code-keyword">this</span>.componentConfig.put(key, value);
    }

    <span class="code-keyword">public</span> void addComponentConfig(<span class="code-object">String</span> key, <span class="code-object">Object</span>[] value) {
<span class="code-keyword">this</span>.componentConfig.put(key, value);
    }
}
</pre>
</div></div>

<p>And our YAML file:</p>

<div class="code panel" style="border-style: solid;border-width: 1px;"><div class="codeHeader panelHeader" style="border-bottom-width: 1px;border-bottom-style: solid;"><b>topology.yaml</b></div><div class="codeContent panelContent">
<pre class="code-java">bolts:
- className: org.apache.storm.flux.wrappers.bolts.FluxShellBolt
  configMethods:
  - name: addComponentConfig
    args: [rabbitmq.configfile, etc/rabbit.yml]
  - name: addComponentConfig
    args:
    - publisher.data_paths
    - [actions]
    
... (<span class="code-keyword">rest</span> of yaml file)
</pre>
</div></div>

<p>It works fine, however it produces this type of warning:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">WARN  o.a.s.f.FluxBuilder - Found multiple invokable methods <span class="code-keyword">for</span> <span class="code-keyword">class </span><span class="code-keyword">class </span>org.apache.storm.flux.wrappers.bolts.FluxShellBolt, method addComponentConfig, given arguments [publisher.data_paths, [actions]]. Using the last one found.
</pre>
</div></div>

<p>Which fortunately happens to be correct method, but it is not correct solution.</p>

<p>Any ideas?</p>

<p>It is quite needed to provide custom config to ShellSpout/Bolt, since we run all spouts/bolts in python via this option. It would be nice to have this option in official release.</p>

---
<details><summary><i>Originally reported by <a title="Pavel Grochal" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=darkless">darkless</a>, imported from: <a href="https://issues.apache.org/jira/browse/STORM-1870" target="_blank">Allow FluxShellBolt/Spout set custom "componentConfig" via yaml.</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Jungtaek Lim" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kabhwan">kabhwan</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2016-10-01T05:10:41+01:00
<li><b>imported</b>: 2025-01-24
</ul></i>
</details>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kabhwan">kabhwan</a>:</i>
<p><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=darkless" class="user-hover" rel="darkless">Pavel Grochal</a><br/>
Sorry it seems a bit late to visit. I found this while taking a look at 'waiting on upstream' label on streamparse.<br/>
Patch is ready for reviewing.</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kabhwan">kabhwan</a>:</i>
<p>Merged into master and 1.x branch.</p>
